eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -136,14 +136,51 @@ jobs:
               $outputPath = Join-Path -Path $(Build.ArtifactStagingDirectory) packages/${{ artifact.Name }}/package-info.json
 
               $version = eng/scripts/Get-PkgVersion -ServiceDirectory ${{ parameters.ServiceDirectory }} -PackageName ${{ artifact.Path }}
-              $outputObject = @{ version = $version.ToString(); name = '${{ artifact.Name }}' } | ConvertTo-Json
+              $outputObject = @{ version = $version.ToString(); name = '${{ artifact.Name }}'; packageName = "${{ artifact.Name }}_$version"; } | ConvertTo-Json
 
               Set-Content -Path $outputPath -Value $outputObject
 
               # Set AZURE_SDK_VERSION for use in Doxygen generation
               Write-Host "##vso[task.setvariable variable=AZURE_SDK_VERSION]$version"
             displayName: Create package info JSON file
 
+          # Building from the root of the repo at this point because of the
+          # nature of the CMake build system. The SoureDirectory should probably
+          # change.
+          - task: powershell@2
+            inputs:
+              targetType: filePath
+              filePath: eng/scripts/New-ReleaseArtifact.ps1
+              arguments: >-
+                -SourceDirectory ./
+                -PackageSpecPath $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/package-info.json
+                -Destination $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/vcpkg
+              pwsh: true
+            displayName: Compress publish artifact and stage for publishing
+
+          - task: Powershell@2
+            inputs:
+              targetType: inline
+              script: |
+                New-Item `
+                  -ItemType Directory `
+                  $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/vcpkg/port -Force
+
+                Copy-Item `
+                  sdk/${{ parameters.ServiceDirectory }}/${{ artifact.Name }}/vcpkg/* `
+                  $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/vcpkg/port `
+                  -Exclude README.md
+
+                Write-Host "Output artifacts to rename"
+                Get-Item $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/vcpkg/port/*.template
+
+                Get-Item $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/vcpkg/port/*.template |
+                  Rename-Item -NewName { $_.Name -replace '.template', '' }
+
+                Get-Item $(Build.ArtifactStagingDirectory)/packages/${{ artifact.Name }}/vcpkg/port/
```

Comment:
Based on our conversation perhaps it is better to do this when you do the template replacement and not during the build stage. Or we just leave them as template in the artifact and rename them after they have been replaced.